### PR TITLE
PieDAO Pie Tokens

### PR DESCRIPTION
**Pull request by MickdeGraaf**
Adds PieDAO Pie Token adapters.

### CODE REVIEW

File: contracts/adapters/pieDAO/PieDAOPieTokenAdapter.sol [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    /**
+     * @return Array of Component structs with underlying tokens rates for the given asset.
+     * @dev Implementation of TokenAdapter interface function.
+     */
+    function getComponents(address token) external view override returns (Component[] memory) {
+        address[] memory underlyingTokensAddresses = IPieSmartPool(token).getTokens();
+        uint256 totalSupply = ERC20(token).totalSupply();
+
```
Comment (sobolev-igor) [inline contracts/adapters/pieDAO/PieDAOPieTokenAdapter.sol:76]:
```suggestion
address bPool = IPieSmartPool(token).getBPool();

```

File: contracts/adapters/pieDAO/PieDAOPieTokenAdapter.sol [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        Component[] memory underlyingTokens = new Component[](underlyingTokensAddresses.length);
+
```
Comment (sobolev-igor) [inline contracts/adapters/pieDAO/PieDAOPieTokenAdapter.sol:78]:
```suggestion

        address underlyingToken;

File: contracts/adapters/pieDAO/PieDAOPieTokenAdapter.sol [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        for (uint256 i = 0; i < underlyingTokens.length; i++) {
+            address underlyingToken = underlyingTokensAddresses[i];
+            underlyingTokens[i] = Component({
+                token: underlyingToken,
+                tokenType: "ERC20",
+                rate: BPool(IPieSmartPool(token).getBPool()).getBalance(underlyingToken) * 1e18 / totalSupply
```
Comment (sobolev-igor) [inline contracts/adapters/pieDAO/PieDAOPieTokenAdapter.sol:84]:
```suggestion
                rate: BPool(bPool).getBalance(underlyingToken) * 1e18 / totalSupply

File: contracts/adapters/pieDAO/PieDAOPieTokenAdapter.sol [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+     * @dev Implementation of TokenAdapter interface function.
+     */
+    function getComponents(address token) external view override returns (Component[] memory) {
+        address[] memory underlyingTokensAddresses = IPieSmartPool(token).getTokens();
+        uint256 totalSupply = ERC20(token).totalSupply();
+
+        Component[] memory underlyingTokens = new Component[](underlyingTokensAddresses.length);
+
+        for (uint256 i = 0; i < underlyingTokens.length; i++) {
+            address underlyingToken = underlyingTokensAddresses[i];
```
Comment (sobolev-igor) [inline contracts/adapters/pieDAO/PieDAOPieTokenAdapter.sol:80]:
```suggestion
            underlyingToken = underlyingTokensAddresses[i];

File: contracts/adapters/pieDAO/PieDAOPieTokenAdapter.sol (hunk 2 above)
Comment (sobolev-igor) [reply contracts/adapters/pieDAO/PieDAOPieTokenAdapter.sol:78]: the idea here is not to create new variable in the loop

File: contracts/adapters/pieDAO/PieDAOPieTokenAdapter.sol (hunk 3 above)
Comment (sobolev-igor) [reply contracts/adapters/pieDAO/PieDAOPieTokenAdapter.sol:84]: and here idea is not to call IPieSmartPool(token) in the loop

File: test/PieDAOPieAdapter.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+        from: accounts[0],
+        gas: '1000000',
+      });
+  });
+
+  it('should return correct balances', async () => {
+    await adapterRegistry.methods['getBalances(address)'](testAddress)
+      .call()
+      .then((result) => {
+        // TODO fix test
```
Comment (sobolev-igor) [inline test/PieDAOPieAdapter.js:106]:
```suggestion

**Comment by MickdeGraaf** [comment]
Should be fixed. Thanks!

**Pull request merged**